.evg.yml

```diff
@@ -320,6 +322,104 @@ tasks:
           content_type: application/octet-stream
           permissions: public-read
 
+  - name: release_tag
+    depends_on:
+      - name: lint
+      - name: test_unit
+      - name: test_with_cloud
+      - name: test_e2e
+      - name: build_publish_clis
+    patchable: false
+    commands:
+      - func: "setup_node"
+      - func: "setup_project"
+      - command: s3.get
+        params:
+          aws_key: ${artifacts_aws_key}
+          aws_secret: ${artifacts_aws_secret}
+          remote_file: versions/cloud-prod/CURRENT
+          bucket: realm-clis
+          local_file: manifest.json
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            set -e
+            cd realm-cli
+            if [ "${branch_name}" != "master" ]; then
+              echo "skipping release: not on master branch"
+              exit 0
+            fi
+            if [ $(git tag -l "v${cli_version}") ]; then
+              echo "skipping release: tag already exists"
+              exit 0
+            fi
+            touch release.true
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            set -e
+            if [ ! -f realm-cli/release.true ]; then
+              # not a release build
+              exit 0
+            fi
+            echo "updating manifest"
+            ./realm-cli/contrib/manifest.js ${build_id} ${cli_version} manifest.json
+      - command: s3.put
+        params:
+          aws_key: ${artifacts_aws_key}
+          aws_secret: ${artifacts_aws_secret}
+          local_file: manifest.json
+          remote_file: versions/cloud-prod/CURRENT
```

Comment:
so I'm understanding, we _always_ push to s3 on every master build, but then we push again on release?

Reply:
yeah, so our s3 interactions currently:

1) on _every_ merge to `master`, we will push CLI builds to S3 (keyed by evergreen's build id ... which is effectively the commit hash)

2) on _every_ merge to `master`, we will fetch AND push the CLI's version manifest (the `CURRENT` doc)

3) on merge to `master` when the `.evg.yml` file has a `cli_version` field which does not yet have a corresponding tag on the remote repo, we will also update the CLI's version manifest to include the new version info (before pushing it back to `s3`)